provider/provider.go

```diff
@@ -56,12 +57,18 @@ func (ap *AivenProvider) Provision(ctx context.Context, provisionData ProvisionD
 }
 
 func (ap *AivenProvider) Deprovision(ctx context.Context, deprovisionData DeprovisionData) (operationData string, err error) {
-	_, err = ap.Client.DeleteService(&aiven.DeleteServiceInput{
+	_, statusCode, err := ap.Client.DeleteService(&aiven.DeleteServiceInput{
 		ServiceName: buildServiceName(ap.Config.ServiceNamePrefix, deprovisionData.InstanceID),
 	})
+
+	if statusCode == http.StatusNotFound {
```

Comment:
It might be worth adding a brief comment here to explain why we're not following the standard pattern of checking for `err != nil` first.

Reply:
That's a code smell. I want to refactor this. I don't think we really need three return values here, especially considering the response body never gets used. The outcomes of the delete method can return success or three kinds of error:

1. network error
2. unexpected response code
3. instance doesn't exist (404)

Perhaps the method should just return an `error`.

Reply:
I'll take a look at it.

Reply:
I've refactored. @jpluscplusm you'll be interested in this, because I went with what I think was your suggestion yesterday.